packages/twenty-front/src/modules/object-record/record-field/form-types/components/FormDateishFieldInputBase.tsx

```diff
@@ -0,0 +1,378 @@
+import { FormFieldInputContainer } from '@/object-record/record-field/form-types/components/FormFieldInputContainer';
+import { FormFieldInputInputContainer } from '@/object-record/record-field/form-types/components/FormFieldInputInputContainer';
+import { FormFieldInputRowContainer } from '@/object-record/record-field/form-types/components/FormFieldInputRowContainer';
+import { VariableChip } from '@/object-record/record-field/form-types/components/VariableChip';
+import { VariablePickerComponent } from '@/object-record/record-field/form-types/types/VariablePickerComponent';
+import { InputLabel } from '@/ui/input/components/InputLabel';
+import {
+  InternalDatePicker,
+  MONTH_AND_YEAR_DROPDOWN_ID,
+  MONTH_AND_YEAR_DROPDOWN_MONTH_SELECT_ID,
+  MONTH_AND_YEAR_DROPDOWN_YEAR_SELECT_ID,
+} from '@/ui/input/components/internal/date/components/InternalDatePicker';
+import { MAX_DATE } from '@/ui/input/components/internal/date/constants/MaxDate';
+import { MIN_DATE } from '@/ui/input/components/internal/date/constants/MinDate';
+import { parseDateToString } from '@/ui/input/components/internal/date/utils/parseDateToString';
+import { parseStringToDate } from '@/ui/input/components/internal/date/utils/parseStringToDate';
+import { useDropdown } from '@/ui/layout/dropdown/hooks/useDropdown';
+import { OverlayContainer } from '@/ui/layout/overlay/components/OverlayContainer';
+import { useListenClickOutside } from '@/ui/utilities/pointer-event/hooks/useListenClickOutside';
+import { UserContext } from '@/users/contexts/UserContext';
+import { isStandaloneVariableString } from '@/workflow/utils/isStandaloneVariableString';
+import { css } from '@emotion/react';
+import styled from '@emotion/styled';
+import {
+  ChangeEvent,
+  KeyboardEvent,
+  useContext,
+  useId,
+  useRef,
+  useState,
+} from 'react';
+import { isDefined, Nullable, TEXT_INPUT_STYLE } from 'twenty-ui';
+
+const StyledInputContainer = styled(FormFieldInputInputContainer)`
+  display: grid;
+  grid-template-columns: 1fr;
+  grid-template-rows: 1fr 0px;
+  overflow: visible;
+  position: relative;
+`;
+
+const StyledDateInputAbsoluteContainer = styled.div`
+  position: absolute;
+`;
+
+const StyledDateInput = styled.input<{ hasError?: boolean }>`
+  ${TEXT_INPUT_STYLE}
+
+  ${({ hasError, theme }) =>
+    hasError &&
+    css`
+      color: ${theme.color.red};
+    `};
+`;
+
+const StyledDateInputContainer = styled.div`
+  position: relative;
+  z-index: 1;
+`;
+
+type DraftValue =
+  | {
+      type: 'static';
+      value: string | null;
+      mode: 'view' | 'edit';
+    }
+  | {
+      type: 'variable';
+      value: string;
+    };
+
+type FormDateishFieldInputBaseProps = {
+  mode: 'date' | 'datetime';
+  label?: string;
+  placeholder?: string;
+  defaultValue: string | undefined;
+  onPersist: (value: string | null) => void;
+  VariablePicker?: VariablePickerComponent;
+};
+
+export const FormDateishFieldInputBase = ({
```

Comment:
This naming is not very clear. See my global comment
```suggestion
export const FormDateTimeFieldInput = ({
```